fe/fe-core/src/main/java/com/starrocks/server/OlapTableFactory.java

```diff
@@ -856,4 +828,45 @@ private void processConstraint(
         olapTable.setForeignKeyConstraints(foreignKeyConstraints);
     }
 
+    private void processFlatJsonConfig(Map<String, String> properties, OlapTable table, String tableName)
+            throws DdlException {
+        if (properties == null || !hasFlatJsonProperties(properties)) {
+            return;
+        }
+
+        try {
+            boolean enableFlatJson = PropertyAnalyzer.analyzeBooleanProp(properties,
+                    PropertyAnalyzer.PROPERTIES_FLAT_JSON_ENABLE, false);
+
+            if (!enableFlatJson) {
+                throw new DdlException("flat JSON configuration must be set after enabling flat JSON.");
```

Comment:
need to re-phrase about this exception message, e.g. `flat_json.enable` must be set to `true` in order to set flat-json related configurations.